### PR TITLE
Remove TestData from series-tests test_api.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
Part of #22550

* Replaced TestData usage in `pandas/tests/series/test_api.py` with fixtures

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/series/test_api.py [hunk 1]
```diff
@@ -39,82 +37,82 @@ def _assert_series_equal(self, left, right):
...
-        assert result.name == self.ts.name
+        result = datetime_series[5:10]
+        assert result.name == datetime_series.name
 
-    def test_pickle(self):
-        unp_series = self._pickle_roundtrip(self.series)
-        unp_ts = self._pickle_roundtrip(self.ts)
-        assert_series_equal(unp_series, self.series)
-        assert_series_equal(unp_ts, self.ts)
+    def test_pickle(self, datetime_series, string_series):
```
Comment (simonjayhawkins) [inline pandas/tests/series/test_api.py]: can you split this test

File: pandas/tests/series/test_api.py [hunk 2]
```diff
@@ -307,44 +305,46 @@ def test_not_hashable(self):
...
-        tm.assert_almost_equal(self.ts.values, self.ts, check_dtype=False)
+    def test_values(self, datetime_series):
+        tm.assert_almost_equal(
+            datetime_series.values, datetime_series, check_dtype=False
+        )
 
-    def test_iteritems(self):
-        for idx, val in self.series.iteritems():
-            assert val == self.series[idx]
+    def test_iteritems(self, datetime_series, string_series):
```
Comment (simonjayhawkins) [inline pandas/tests/series/test_api.py]: and again

File: pandas/tests/series/test_api.py [hunk 3]
```diff
@@ -307,44 +305,46 @@ def test_not_hashable(self):
...
 
-    def test_contains(self):
-        tm.assert_contains_all(self.ts.index, self.ts)
+    def test_contains(self, datetime_series):
+        tm.assert_contains_all(datetime_series.index, datetime_series)
 
-    def test_iter(self):
-        for i, val in enumerate(self.series):
-            assert val == self.series[i]
+    def test_iter(self, datetime_series, string_series):
```
Comment (simonjayhawkins) [inline pandas/tests/series/test_api.py]: and this one.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@SaturnFromTitan thanks for the PR.

can you split the tests where the two fixtures are tested independently

File: pandas/tests/series/test_api.py [hunk 4]
```diff
@@ -307,44 +307,49 @@ def test_not_hashable(self):
...
 
-        for idx, val in self.ts.items():
-            assert val == self.ts[idx]
+    def test_items_strings(self, string_series):
+        for idx, val in string_series.items():
+            assert val == string_series[idx]
 
         # assert is lazy (genrators don't define reverse, lists do)
-        assert not hasattr(self.series.items(), "reverse")
+        assert not hasattr(string_series.items(), "reverse")
```
Comment (SaturnFromTitan) [inline pandas/tests/series/test_api.py]: I don't get why this is part of the test, but I left it as it was there before. Currently, it's only part of the string_series tests for `.items` and `.iteritems`, but not for the datetime_series ones (same as it was before).

@simonjayhawkins What's your take on removing it vs leaving as is?

Comment (simonjayhawkins) [reply pandas/tests/series/test_api.py]: I think fine to leave as is.

**Review by simonjayhawkins — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @SaturnFromTitan very nice cleanup

**Pull request merged**